jupyter_packaging/setupbase.py

```diff
@@ -63,17 +64,18 @@ def list2cmdline(cmd_list):
 # ---------------------------------------------------------------------------
 
 
-def get_data_files(top):
-    """Get data files"""
+def expand_data_files(data_file_patterns):
+    """Expand data file patterns to a valid data_files spec.
 
+    Parameters
+    -----------
+    data_file_patterns: list(tuple)
+        A list of (directory, glob pattern) for the data file locations.
+    """
     data_files = []
-    ntrim = len(here + os.path.sep)
-
-    for (d, _, filenames) in os.walk(top):
-        data_files.append((
-            d[ntrim:],
-            [pjoin(d, f) for f in filenames]
-        ))
+    for (directory, pattern) in data_file_patterns:
+        files = [os.path.relpath(f, here) for f in glob(pattern)]
```

Comment:
Will this recurse into subdirectories, or will users have to include one pattern per subdir?

Reply:
`glob` itself can recurse with `**`, but we could allow for more flexibility by accepting a list of globs.

Reply:
I've burnt myself on this before, so I double checked the docs (https://docs.python.org/3/library/glob.html#glob.glob). Support for the `**` pattern was only added in Python 3.5, and is hidden behind a flag `recursive`.